### ISSUE TITLE
Unable to locate `DeprecationWarning: In future, it will be an error for 'np.bool_' scalars to be interpreted as an index`

### ISSUE THREAD

**Issue by albertotb** [OceanParcels/parcels#1119]
I'm getting this `DeprecationWarning`:

```
DeprecationWarning: In future, it will be an error for 'np.bool_' scalars to be interpreted as an index
    self.cstruct = CStructuredGrid(self.xdim, self.ydim, self.zdim,
```

I cannot find it in the source code and thus I cannot find what piece of code is triggering the warning. The only thing I can think of is this

```
VARIABLES = {"U": "uo", "V": "vo"}
DIMENSIONS = {"lat": "latitude", "lon": "longitude", "time": "time", "depth": "depth"}

fieldset = parcels.FieldSet.from_netcdf(
            fnames,
            VARIABLES,
            DIMENSIONS,
            indices=indices,
            mesh="spherical",
            allow_time_extrapolation=True,
)
```

but I do not see any bools around, since `indices` is a dict of list of scalars for each dimension. I'm worried that my code will stop working in future versions. Can someone point me in the right direction?

**Comment by erikvansebille** [comment]
Yes, this is an annoying Warning that has been around for a while. I looked into it earlier, but couldn't find an easy way to fix it. It seems related to these lines
https://github.com/OceanParcels/parcels/blob/0df79e6cea94913e1b1565443e653fe72e87de62/parcels/grid.py#L120-L130
@Ckehl, perhaps you know what's causing this Warning and how to fix it?

For now, it doesn't have an impact on Parcels (yet), but something to fix soon(ish) for sure, before numpy deprecates this feature

**Comment by albertotb** [comment]
Ok, thanks for taking the time to look into this. I thought that it was something that I was doing wrong in my code. I think the change was made in numpy version 1.14 (https://docs.scipy.org/doc/numpy-1.14.0/release.html):

- Using np.bool_ objects in place of integers is deprecated. Previously operator.index(np.bool_) was legal and allowed constructs such as [1, 2, 3][np.True_]. That was misleading, as it behaved differently from np.array([1, 2, 3])[np.True_].

It seems it is also happening in other libraries like https://github.com/skyfielders/python-skyfield/issues/486

**Comment by CKehl** [comment]
The issue comes from `self.z4d`.

Generally for a Grid, we set this variable as an integer value (grid.py, l. 336):
```
self.z4d = -1  # only used in RectilinearSGrid
```

But for S-grids (RectilinearSGrid, l. 371; CurvilinearSGrid, l.500), it is set as a boolean value:
```
self.z4d = len(self.depth.shape) == 4
```

Therefore, this warning is emitted for SGrids (as far as I can see it now), because of the required implicit-cast from boolean to integer. It could be as easy as writing (for SGrids) instead:
```
self.z4d = int(len(self.depth.shape) == 4)
```

or

```
self.z4d = 1 if len(self.depth.shape) == 4 else 0
```

**Comment by erikvansebille** [comment]
Ah thanks @CKehl! I just created a PR with this change #1123

**Comment by erikvansebille** [comment]
Fixed in #1123
